src/util/memory.h

```diff
@@ -52,13 +52,13 @@
    #elif defined(__GNUC__)
       // std::make_unique was added in gcc 4.9
       #define GCC_VERSION (__GNUC__ * 10000 + __GNUC_MINOR__ * 100 + __GNUC_PATCHLEVEL__)
-      #if GCC_VERSION >= 40900
+      #if GCC_VERSION >= 40900 && __cplusplus > 201103L
```

Comment:
Sorry, missed the mention in the PR description. So this would prevent defining that the compiler supports make_unique when the user specified -std=c++04 for example?